### PR TITLE
Fix home/base installation

### PR DESCRIPTION
**Pull request by hunner**
Tomcat is traditionally installed into catalina_home, and multiple
instances can be created by using a custom catalina_base per instance.
The previous iteration of this was buggy, so this commit implements a
new defined resource tomcat::install which is used for declaring each
install of the software into a catalina_home, and then tomcat::instance
can deploy an instance from catalina_home to catalina_base and declare a
tomcat::service to go along with it.

The base class is used to set a few defaults to be used by all
instances.

If the old behavior is to be used, this is still possible.

### CODE REVIEW

**Comment by tphoney** [comment]
wow, :+1:

File: manifests/service.pp [hunk 1]
```diff
@@ -28,53 +28,49 @@
...
   $service_enable = undef,
-  $use_init       = false,
   $service_name   = undef,
   $start_command  = undef,
   $stop_command   = undef,
-  $user           = $::tomcat::user,
+  $user           = undef,
 ) {
+  include tomcat
+  $_user = pick($user, $::tomcat::user)
```
Comment (bmjen) [inline manifests/service.pp]: Is this pick necessary? It looks just just another way to default `$user` to `$::tomcat::user`.

Comment (hunner) [inline manifests/service.pp]: The `tomcat::service` define cannot inherit the tomcat class, so `$::tomcat::user` may not be available at parameter default time so it must be brought into the definition; you can see the `include tomcat` line right above this.

**Comment by bmjen** [comment]
:+1: :100:

**Comment by hunner** [comment]
Ran it through CI; 107 examples, 33 failures :(

**Pull request merged**